### ISSUE TITLE
Debugging tests doesn't set environment variables

### ISSUE THREAD

**Issue by dadocolussi** [matepek/vscode-catch2-test-adapter#237]
**Checklist**

- [x] The issue is about this extension and NOT about a fork.
- [x] Check the [known issues](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md#known-issues) list.
- [x] The latest version of the extension was used.
- [x] It is not related to remote-vscode or I checked the following [issue](https://github.com/matepek/vscode-catch2-test-adapter/issues/201)

**Describe the bug**

Environment variable set under `advancedExecutables` in `settings.json` is not set when debugging test cases. When running a test case (without debugging it), the environment variable is available as expected. The problem is reproducible only when debugging. The problem is also reproducible locally as well as using remote-ssh.

**To Reproduce**

I have the following `settings.json` entry. I expect it to have the effect of HELLO environment variable being set for the test process:
```
	"testMate.cpp.test.advancedExecutables": [
		{
			"pattern": "MyUnitTestProgram",
			"env": {"HELLO":"Hello World!"}
		}
```
I have the following Catch2 test case that checks the presence of the environment variable. The problem is reproducible with Google Test, too.
```
TEST_CASE("HELLO environment variable is set")
{
	const char *s = ::getenv("HELLO");
	REQUIRE(s != nullptr);
}
```

Debug the test case and it'll fail. Run it normally without debugging, and it'll succeed.

**Desktop**

- Extension Version:3.6.7
- VS Code Version: 1.51.1
- Catch2 / Google Test / DOCTest Version: 2.13.3
- OS Type and Version: macOS 10.15.7
- Using remote-ssh/docker/wsl?: remote-ssh and local.

**Comment by matepek** [comment]
Hello,

Nice bug report. 
Could you tell me which debug adapter are you using?
Or you can send me the logs. 
One line would be really important: `Debug: resolved debugConfig`

**Comment by dadocolussi** [comment]
I'm using `ms-vscode.cpptools`. Is that what you mean?

**Comment by dadocolussi** [comment]
My current workaround is to remove `testMate.cpp.debug.configTemplate` and duplicate the environment variables in the first matching entry in `launch.json` (described in the config template documentation). The environment variables from `launch.json` do make it to the process when debugging. I still need the same variables under `testMate.cpp.test.advancedExecutables` when running the tests without debugging.

With the workaround, I observe in the logs:
* `Using debug { debugConfigTemplateSource: 'fromLaunchJson' }`.
* `Debug: resolved debugConfig:` contains:
    - `environment: [ { name: 'HELLO', value: 'Hello World!' } ]`, and this is visible in the process. 
    - `env` contains `HELLO: 'Hello World!'` and this is not visible in the process.

**Comment by matepek** [comment]
It would have been interesting also that how the log looks like in case of the original repro/issue.

You've probably already read the [documentation](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/configuration/debug.configTemplate.md).

Well there is always the option to fill your own debug config. The mechanism I created is merely a fallback. Anyhow I tried to fix it. Check the next release.
